pkg/controller/linstorcontrollerset/linstorcontrollerset_controller.go

```diff
@@ -300,40 +270,20 @@ func (r *ReconcileLinstorControllerSet) Reconcile(request reconcile.Request) (re
 		return reconcile.Result{}, err
 	}
 
-	logrus.WithFields(logrus.Fields{
-		"name":      ctrlSet.Name,
-		"namespace": ctrlSet.Namespace,
-	}).Debug("CS Reconcile: StatefulSet already exists")
+	log.Debug("CS Reconcile: StatefulSet already exists")
 
-	errs := r.reconcileControllers(pcs)
-	compoundErrorMsg := newCompoundErrorMsg(errs)
-	pcs.Status.Errors = compoundErrorMsg
+	errs := r.reconcileControllers(ctx, pcs)
 
-	if err := r.client.Status().Update(context.TODO(), pcs); err != nil {
-		logrus.WithFields(logrus.Fields{
-			"err": err,
-		}).Error("CS Reconcile: Failed to update LinstorControllerSet status")
-		for _, err := range errs {
-			logrus.WithFields(logrus.Fields{
-				"err": err,
-			}).Error("CS Reconcile: Error could not be added to LinstorControllerSet status")
-		}
-		return reconcile.Result{}, err
-	}
-
-	logrus.WithFields(logrus.Fields{
-		"errs": compoundErrorMsg,
-	}).Debug("reconcile loop end")
-
-	var compoundError error
-	if len(compoundErrorMsg) != 0 {
-		compoundError = fmt.Errorf("CS Reconcile: requeuing reconcile loop for the following reason(s): %s", strings.Join(compoundErrorMsg, " ;; "))
+	statusErr := r.reconcileStatus(ctx, pcs, errs)
+	if statusErr != nil {
+		log.Warnf("failed to update status. original errors: %v", errs)
+		return reconcile.Result{}, statusErr
 	}
 
-	return reconcile.Result{RequeueAfter: time.Minute * 1}, compoundError
+	return reconcileutil.ToReconcileResult(errs...)
```

Comment:
Maybe log something to indicate what we are returning here? I don't think we can see that otherwise if there are no actual errors returned. Same for node set controller.